src/Iterators/CsvIterator.php

```diff
@@ -47,12 +47,13 @@ public function current()
             throw new IncorrectCSVFileException('Incorrect CSV file');
         }
 
-        return array_associate($this->currentCsvLine, function ($value, $key) {
-            return [
-                'key' => $this->columns[$key],
-                'value' => $value
-            ];
+        $result = [];
+
+        array_walk($this->currentCsvLine, function($value, $key) use (&$result) {
```

Comment:
such logic wouldn't remove legacy keys, let's use mapWithKeys helper

```php
collect($this->currentCsvLine)->mapWithKeys(function ($item) {
    return [$item['email'] => $item['name']];
});
```

Reply:
@DenTray There is no legacy keys. getcsv() returns indexed array

Reply:
@DenTray done with array_combine